nixos/doc/manual/release-notes/rl-2009.xml

```diff
@@ -61,6 +61,19 @@
       This is to make it possible to use <literal>podman</literal> instead of <literal>docker</literal>.
     </para>
    </listitem>
+   <listitem>
+    <para>
+      MariaDB has been updated to 10.4, MariaDB Galera to 26.4.
+      Before you upgrade, it would be best to take a backup of your database.
+      For MariaDB Galera Cluster, see <link xlink:href="https://mariadb.com/kb/en/upgrading-from-mariadb-103-to-mariadb-104-with-galera-cluster/">Upgrading
+      from MariaDB 10.3 to MariaDB 10.4 with Galera Cluster</link> instead.
+      Before doing the upgrade read <link xlink:href="https://mariadb.com/kb/en/upgrading-from-mariadb-103-to-mariadb-104/#incompatible-changes-between-103-and-104">Incompatible
+      Changes Between 10.3 and 10.4</link>.
+      MariaDB 10.4 introduces a number of changes to the authentication process, intended to make things easier and more
+      intuitive. See <link xlink:href="https://mariadb.com/kb/en/authentication-from-mariadb-104/">Authentication from MariaDB 10.4</link>.
```

Comment:
As a NixOS user upgrading to 20.09, I still don't grok what these "number of changes to the authentication process" now effectively mean for me.

I assume I can't do `sudo -u testuser mysql -u testuser -e 'use testdb; select test_id from db1;' -N` anymore, which is why you updated the test. My question is, was this already the case with mariadb 10.3.22?

Reply:
People need to be able to reason about the changes being done, and implications this may have on their configuration.

Reply:
These changes appeared in 10.4.
Not worked this command:
```
sudo -u testuser mysql -u root
```
Error:
```
ERROR 1698 (28000): Access denied for user 'root'@'localhost'
```
By default, `mysql_install_db` creates two user and used `unix_socket` authentication:
```
CREATE USER root@localhost IDENTIFIED VIA unix_socket OR mysql_native_password USING 'invalid'
CREATE USER mysql@localhost IDENTIFIED VIA unix_socket OR mysql_native_password USING 'invalid'
```

Reply:
"It doesn't work anymore, but if I change it to $this, it works again" isn't a sufficient explanation, sorry.

We need to assess the changes and the implications this has on people making use of it, and giving an appropriate heads-up in the release notes at least. If the tests needed updates, it's likely users configuration need changes, too.

Reply:
When updating, the old authentication method is used. With a new installation it is used `unix_socket` authentication.

Reply:
@flokli update release notes.